src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -121,6 +144,204 @@ public void importRepositories(final ProgrammingExercise templateExercise, final
+            if (testRepo) {
+                repositoryService.createFile(repository, "test/" + file.getFileName(), Files.newInputStream(file));
+            }
+            else {
+                repositoryService.createFile(repository, "src/" + file.getFileName(), Files.newInputStream(file));
+            }
+
+        }
+    }
+
+    private void copyExerciseContentToRepositoryForJavaOrKotlinExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir, ProgrammingExercise newExercise,
+            String oldPackageName) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "test/" + newExercise.getPackageFolderName());
+            testRepo = true;
+        }
+        else {
+            testRepo = false;
+            repositoryService.createFolderRecursively(repository, "src/" + newExercise.getPackageFolderName());
+        }
+        Map<String, String> replacePackageName = Map.of(oldPackageName, newExercise.getPackageName());
+
+        retrieveRepositoryContentPathsJavaKotlin(exerciseDir, repositoryType).forEach(file -> {
+            try {
+                if (testRepo) {
+                    repositoryService.createFile(repository, "test/" + newExercise.getPackageFolderName() + "/" + file.getFileName(), Files.newInputStream(file));
+                }
+                else {
+                    repositoryService.createFile(repository, "src/" + newExercise.getPackageFolderName() + "/" + file.getFileName(), Files.newInputStream(file));
+                }
+                var path = repository.getLocalPath();
+                path = testRepo ? path.resolve("test") : path.resolve("src");
+                var absolutePath = path.toAbsolutePath();
+                fileService.replaceVariablesInFileRecursive(absolutePath.toString(), replacePackageName);
+            }
+            catch (IOException e) {
+                throw new InternalServerErrorException("Could not copy exercise content to repository");
+            }
+
+        });
+    }
+
+    private void deleteSourceAndTestFilesInRepository(Repository repository) {
+        // only the root level dir exists, delete all existing files
+        if (gitService.listFilesAndFolders(repository).entrySet().stream().filter(e -> e.getValue() == FileType.FOLDER).count() == 1) {
+            gitService.listFilesAndFolders(repository).entrySet().stream().filter(e -> FileType.FILE == e.getValue()).forEach(e -> {
+                if (!FileSystemUtils.deleteRecursively(e.getKey())) {
+                    log.error("failed to delete: " + e.getKey().getName());
+                }
+            });
+            repository.setContent(null);
+            return;
+        }
+        Stream.of("src", "test", "behavior", "structural", "assignment", "checker", "overrides", "solution", "tests", "testUtils")
```

Comment:
Why is this special filter needed? Shouldn’t the repository be cleaned up fully on import so that it only contains the imported files, i.e. has the same state as the original exercise repository?

Reply:
This by extension also brings up another question: Why are all the copy mechanisms specific per programming language/project type needed? Why is it not enough to copy the content of the zip 1:1 to the new repository?

Reply:
no longer part of the implementation